src/mlpack/methods/linear_regression/linear_regression_main.cpp

```diff
@@ -11,6 +11,12 @@
  */
 #include <mlpack/prereqs.hpp>
 #include <mlpack/core/util/io.hpp>
+
+#ifdef BINDING_NAME
+  #undef BINDING_NAME
+#endif
+#define BINDING_NAME linear_regression
```

Comment:
The changes to the binding itself are not too intrusive.  We now just need to set `BINDING_NAME` appropriately in the beginning, and all calls like `IO::HasParam()` are now `params.Has()`, etc.  I think it could be possible to just regex the rest of the bindings and get good results.